lib/private/Files/View.php

```diff
@@ -380,6 +378,24 @@ public function rmdir($path) {
 		}, ['before' => ['path' => $this->getAbsolutePath($path)], 'after' => ['path' => $this->getAbsolutePath($path)]], 'file', 'delete');
 	}
 
+	/**
+	 * Checks whether given path is a share folder or one of share folder parents
+	 *
+	 * @param $path - path relative to the files folder
+	 *
+	 * @return bool
+	 */
+	protected function isShareFolderOrShareFolderParent($path) {
+		$shareFolder = \trim($this->config->getSystemValue('share_folder', '/'), '/');
+		if ($shareFolder === '') {
+			return false;
+		}
+		$user = $user = \OC_User::getUser();
+		$shareFolderAbsolutePath = "/$user/files/$shareFolder";
+		$trimmedAbsolutePath = $this->getAbsolutePath(\trim($path, '/'));
+		return $shareFolderAbsolutePath === $trimmedAbsolutePath || \strpos($shareFolderAbsolutePath, "$trimmedAbsolutePath/") === 0;
```

Comment:
```suggestion
		return $shareFolderAbsolutePath === $trimmedAbsolutePath || \strpos("$shareFolderAbsolutePath/", "$trimmedAbsolutePath/") === 0;
```
without this, I think that if `share_folder` is `Shares` and someone also creates their own folder called `SharesMyLocalExtraFolder` then they will be prevented from deleting `SharesMyLocalExtraFolder`

Reply:
Now I see that this does not matter. The first `===` test handles the case when the action is on the actual share folder. So the 2nd test is only needed when the `$trimmedAbsolutePath` is a parent of the share folder. And in that case it does not matter if `$shareFolderAbsolutePath` has a trailing `/` or not.
I tried a few combinations, and it works OK (and CI is passing)

Reply:
@phil-davis trailing slash matters. Otherwise it won't be possible to delete 
`/sharedfiles`  when  the share_folder is set to `/shared`
We have unit tests for that btw :)
https://github.com/owncloud/core/blob/master/tests/lib/Files/ViewTest.php#L2726